lib/mapped.js

```diff
@@ -30,9 +30,9 @@ MappedCoverage.prototype.addStatement = function (loc, hits) {
         meta = this.meta,
         index = meta.seen[key];
 
-    if (!index) {
-        meta.last.s += 1;
+    if (index === undefined) {
         index = meta.last.s;
+        meta.last.s += 1;
```

Comment:
Let me see if I'm following here:

1. there was logic in place that handled populating a statement the first time that it was observed; this logic used `!` so was triggering on `0`, which currently counts as a valid mapping.
2. as a result, we accidentally fall outside of bounds in `istanbul-lib-coverage`? could you better explain to me the cause of this?

This fix addresses the issue by looking for `undefined` rather than `0`.